src/classes/Gift.cls

```diff
@@ -47,10 +48,16 @@ public with sharing class Gift {
         }
         set;
     }
-    
+
+    public Gift(Id giftId) {
+        this.dataImport = giftSelector.getGiftById(giftId, new GiftEntryFormTemplate());
```

Comment:
It seems a Gift Entry Gift (or maybe just a Batch) is dependent on a Template, since it determines its View state.  What would you think about renaming  `GiftEntryFormTemplate` to `GiftTemplate` ?

Our language could be "The Form uses the Batch's Gift Template (or the default Gift Template in SGE) to display Gifts on the Form."

It also makes more sense for the GiftSelector to get its fields from a GiftTemplate than a "Form"Template, which is tied to a specific UI.

So, I think the Batch should know about the Template (to configure the Selector), but then the Batch should configure the Selector and pass that into the constructor of the Gifts instance.